resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/VolcanoTestsSuite.scala

```diff
@@ -63,8 +150,40 @@ private[spark] trait VolcanoTestsSuite { k8sSuite: KubernetesSuite =>
       }
     )
   }
+
+  test("Run 4 SparkPi Jobs with 2 volcano queues (queue scheduling)", k8sTestTag, volcanoTag) {
+    createOrReplaceYAMLResource(VOLCANO_Q0_DISABLE_Q1_ENABLE_YAML)
+    val jobNum = 4
+    val groupLocator = "queue-test" + UUID.randomUUID().toString.replaceAll("-", "")
+    // Submit two jobs into disabled queue0 and enabled queue1
+    (1 to jobNum).foreach { i =>
+      Future {
+        runJobAndVerify(i.toString, Option(groupLocator), Option(s"queue${i % 2}"))
+      }
+    }
+    // There are two `Succeeded` jobs and two `Pending` jobs
+    Eventually.eventually(TIMEOUT, INTERVAL) {
+      val completedPods = getPods("driver", groupLocator, "Succeeded")
+      assert(completedPods.size === 2)
+      val pendingPods = getPods("driver", groupLocator, "Pending")
+      assert(pendingPods.size === 2)
+    }
```

Comment:
Actually, this is weak because this test will pass in a normal sequential scheduler too. We had better check the content of driver explicitly.

Reply:
If possible, please two group locators. One for Q0 and the other for Q1.  That would be easier.